modules/programs/ssh.nix

```diff
@@ -154,10 +156,20 @@ in
     };
 
     matchBlocks = mkOption {
-      type = types.listOf matchBlockModule;
+      type = types.loaOf matchBlockModule;
       default = [];
+      example = {
+        "john.example.com" = {
+          hostname = "example.com";
+          user = "john";
+        };
+        foo = {
+          hostname = "example.com";
+          identityFile = "/home/john/.ssh/foo_rsa";
+        };
+      };
       description = ''
-        Specify per-host settings.
+        Specify per-host settings. See <citerefentry><refentrytitle>ssh_config</refentrytitle><manvolnum>5</manvolnum></citerefentry>
```

Comment:
Maybe add an emphasis that this must be specified as a list if the order of rules matter. Something like
```
      description = ''
        Specify per-host settings. If the order of rules matter then
        this must be a list. See
        <citerefentry>
          <refentrytitle>ssh_config</refentrytitle>
          <manvolnum>5</manvolnum>
        </citerefentry>.
      '';
```
perhaps.